src/Components/App/App.js

```diff
@@ -27,130 +29,181 @@ class App extends Component {
+
+  gameFilter = (activity, answers) => {
+    let possibleSuggestions = this.state[activity];
+    const choices = ["1", "2", "3", "4", "5", "more than 5"];
+    let numberOfPlayers = choices.find((choice) => {
+      return answers.includes(choice);
+    });
+    return possibleSuggestions.filter((element) => {
+      if (numberOfPlayers === "more than 5") {
+        return element.max_players >= 5;
+      }
+      return (
+        element.min_players <= numberOfPlayers &&
+        element.max_players >= numberOfPlayers
+      );
+    });
+  };
 
   filterActivity = (activity, answers) => {
-    //refactor this - break out into helper functions
-    let genreGroup = ['movies', 'podcasts', 'music']
-    let gamesGroup = ['boardGames', 'cardGames']
-    let ageGroup = ['movies', 'music']
+    let genreGroup = ["movies", "podcasts", "music"];
+    let gamesGroup = ["boardGames", "cardGames"];
+    let ageGroup = ["movies", "music"];
     let possibleSuggestions;
     if (genreGroup.includes(activity)) {
-      possibleSuggestions = this.state[activity].filter(singleActivity => {
-        if (activity === 'movies') {
-          return singleActivity.genre.some(genre => answers.includes(genre))
-        }
-        return answers.some(genre => genre === singleActivity.genre)
-      })
+      possibleSuggestions = this.genreFilter(activity, answers);
     }
     if (ageGroup.includes(activity)) {
-      let ageRestriction = answers.find(answer => {
-        return answer.includes('\'s')
-      })
-      if (ageRestriction) {
-        possibleSuggestions = possibleSuggestions.filter(element => {
-          return element.release_date.split('-')[0] > ageRestriction.split('\'s')[0]
-        })
-      }
+      possibleSuggestions = this.ageFilter(possibleSuggestions, answers);
     }
-    if (activity === 'movies') {
-        if (answers.includes('too long')) {
-          possibleSuggestions = possibleSuggestions.filter(element => {
-            return +element.runtime < 120
-          })
-        }
-        if (answers.includes('that\'s fine')) {
-          return possibleSuggestions
-        }
+    if (activity === "movies") {
+      possibleSuggestions = this.movieFilter(possibleSuggestions, answers);
     }
     if (gamesGroup.includes(activity)) {
-      possibleSuggestions = this.state[activity]
-      const choices = ['1', '2', '3', '4', '5', 'more than 5']
-      let numberOfPlayers = choices.find(choice => {
-        return answers.includes(choice)
-      })
-      possibleSuggestions = possibleSuggestions.filter(element => {
-        if (numberOfPlayers === 'more than 5') {
-          return element.max_players >= 5
-        }
-        return element.min_players <= numberOfPlayers && element.max_players >= numberOfPlayers
-      })
+      if (this.state.cardGames && this.state.boardGames && activity === "boardGames"){
+        answers = this.state.cardGamesAnswers
+      }
+      possibleSuggestions = this.gameFilter(activity, answers);
     }
```

Comment:
Is the logic from 184 - 187 accounting for a scenario when the user chooses both board games and card games?  If so, then we're setting the answers variable to the single answer regarding how many people are playing?

Reply:
So with the way we removed the duplication of the `how many players` questions, we no longer update `boardGamesAnswers` when both a `cardGame` and `boardGame` are selected. The logic here says `IF` both `cardGame` and `boardGames` have been selected, and the current reduce iteration is for the `boardGames`, assign the `answers` to whatever the answer to the `how many players` question, which is stored in `cardGamesAnswers`

Sorry that was a lot of games, but basically this is just a workaround since previously `boardGame` answers would not be populated by our filters if both game types were selected

Reply:
Thanks!